allennlp/training/learning_rate_schedulers.py

```diff
@@ -118,21 +124,101 @@ def step_batch(self, epoch=None):
         for param_group, learning_rate in zip(self.optimizer.param_groups, self.get_lr()):
             param_group['lr'] = learning_rate
 
-
     def get_lr(self):
         step = max(self.last_epoch, 1)
         scale = self.factor *  (self.model_size ** (-0.5) *
                                 min(step ** (-0.5), step * self.warmup_steps ** (-1.5)))
 
         return [scale for _ in range(len(self.base_lrs))]
 
+
+class CosineWithRestarts(torch.optim.lr_scheduler._LRScheduler):  # pylint: disable=protected-access
+    """
+    Cosine annealing with restarts.
+
+    This is decribed in the paper https://arxiv.org/abs/1608.03983.
+
+    Parameters
+    ----------
+    optimizer : torch.optim.Optimizer
+
+    T_max : int
```

Comment:
Can you use a lowercase `t_max` here, too, to match variable naming conventions?